addons/netfox/rollback/network-rollback.gd

```diff
@@ -29,6 +29,19 @@ var display_offset: int:
 	set(v):
 		push_error("Trying to set read-only variable display_offset")
 
+## How many previous input frames to send along with the current one.
+##
+## Input data is sent unreliably over UDP for speed. 
+## Some packets may be lost, some arrive late or out of order.
+## To mitigate this, we can send the current and previous n ticks of input data.
+##
+## [i]read-only[/i], you can change this in the project settings
+var input_redundency: int:
+	get:
+		return ProjectSettings.get_setting("netfox/rollback/input_redundency", 5)
```

Comment:
Purely a query - have you played around with different redundancy values?

Reply:
I had but your comment made me realise I should get some probability data and justify why 5 should be the default. I'm now thinking 3 is a reasonable default value:

At 20% packet loss ( really bad connection ) these would be the success chances.

Redundancy: 0, Success Percentage: 80.00%
Redundancy: 1, Success Percentage: 96.00%
Redundancy: 2, Success Percentage: 99.20%
Redundancy: 3, Success Percentage: 99.84%
Redundancy: 4, Success Percentage: 99.97%
Redundancy: 5, Success Percentage: 99.99%
Redundancy: 6, Success Percentage: 100.00%

Reply:
Thanks for diving deeper! I agree with 3, after that it's diminishing returns.